libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx

```diff
@@ -1,51 +1,87 @@
 // (C) 2022 GoodData Corporation
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import cx from "classnames";
-import { DateRangePickerInputField } from "./DateRangePickerInputField";
 import { injectIntl, WrappedComponentProps } from "react-intl";
-import DayPickerInput from "react-day-picker/DayPickerInput";
 import moment from "moment";
+import isValid from "date-fns/isValid";
+import parse from "date-fns/parse";
+import format from "date-fns/format";
+
 import { DateRangePickerInputFieldBody } from "./DateRangePickerInputFieldBody";
+
 import { convertPlatformDateStringToDate } from "../utils/DateConversions";
 import { TIME_FORMAT } from "../constants/Platform";
-import { DayPickerProps } from "react-day-picker";
 import { getPlatformStringFromDate, getTimeStringFromDate } from "./utils";
 
+function formatDate(date: Date, dateFormat: string): string {
+    return format(date, dateFormat);
+}
+
+function parseDate(str: string, dateFormat: string): Date | undefined {
+    try {
+        const parsedDate: Date = parse(str, dateFormat, new Date());
+        // parse only dates with 4-digit years. this mimics moment.js behavior - it parses only dates above 1900
+        // this is to make sure that the picker input is not overwritten in the middle of writing the year with year "0002" when writing 2020.
+        //
+        // it's also necessary to parse only when the input string fully matches with the desired format
+        // to make sure that the picker input is not overwritten in the middle of writing.
+        // e.g, let's consider a case where dateFormat is "dd/MM/yyyy" and the DayPickerInput has already been filled with a valid string "13/09/2020",
+        // then an user wants to change only the month "13/09/2020" -> "13/11/2020" by removing "09" and typing "11".
+        // in such case the parsing should wait until the user completes typing "11" (otherwise if parsing is done right after the first "1" is typed,
+        // the cursor automatically moves to the end of the string in the middle of writing, causing a bad experience for the user).
+        if (
+            isValid(parsedDate) &&
+            parsedDate.getFullYear() >= 1000 &&
```

Comment:
just curious, why 1000?

Reply:
Not quite sure, was already there. Guess is some other condition to be sure that the input has correct value before starting with the calculations.

Reply:
:+1: